esrally/log.py

```diff
@@ -40,6 +43,36 @@ def configure_utc_formatter(*args, **kwargs):
     return formatter
 
 
+class RallyEcsFormatter(ecs_logging.StdlibFormatter):
+    def __init__(self, *args: typing.Any, **kwargs: typing.Any):
+        super().__init__(*args, **kwargs)
+
+    def format(self, record: logging.LogRecord) -> str:
+        result = super().format_to_ecs(record)
+        self.rename_actor_fields(result)
+        # ecs_logging._utils is a private module
+        # but we need to use it here to get the on spec JSON serialization
+        return ecs_logging._utils.json_dumps(result)  # pylint: disable=protected-access
+
+    def rename_actor_fields(self, log_record: typing.Dict[str, typing.Any]) -> None:
+        actor = {}
+        if log_record.get("actorAddress"):
+            actor["address"] = log_record.pop("actorAddress")
+        if log_record.get("taskName"):
+            actor["task"] = log_record.pop("taskName")
+        if actor:
+            collections.deep_update(log_record, {"rally": {"actor": actor}})
```

Comment:
Or we can just drop `taskName` for now.

Reply:
> Huh, turns out that's fairly new.

Ah, yes, most recently I've tested with 3.11.7 from esbench env. After switching to 3.12.2 and:

```
diff --git a/esrally/resources/logging.json b/esrally/resources/logging.json
index c1e4b6da..7651ebc5 100644
--- a/esrally/resources/logging.json
+++ b/esrally/resources/logging.json
@@ -2,12 +2,12 @@
   "version": 1,
   "formatters": {
     "normal": {
-      "format": "%(asctime)s,%(msecs)d %(actorAddress)s/PID:%(process)d %(name)s %(levelname)s %(message)s",
+      "format": "%(asctime)s,%(msecs)d %(actorAddress)s/PID:%(process)d %(name)s %(taskName)s %(levelname)s %(message)s",
       "datefmt": "%Y-%m-%d %H:%M:%S",
       "()": "esrally.log.configure_utc_formatter"
     },
     "json": {
-      "format": "%(message)s",
+      "format": "%(message)s %(taskName)s",
       "()": "esrally.log.configure_ecs_formatter"
     },
     "profile": {
```

I can see `taskName` populated in `rally.log` but not in `rally.json`. I think it gets swallowed in [here](https://github.com/elastic/ecs-logging-python/blob/a4ef6e1ffff4f41b0f66d52a8dc93e77f6f70ac3/ecs_logging/_stdlib.py#L167). We could override `format_to_ecs()` but I'm also fine just skipping it for now. We can revisit once we find a use for this information. What we're adding in `process` object is already very useful.

Still interested in hearing the rationale behind `deep_update()` :)

Reply:
`deep_update` is so that _just in case_ we have something already there, like someone adds a future mutator, we don't clobber any dict that is already defined. Like would happen with `dict.update()`